tests/tests_geomstats/test_positive_lower_triangular_matrices.py

```diff
@@ -2,287 +2,323 @@
-                Matrices.to_strictly_lower_triangular(Y),
-            )
-            + Matrices.frobenius_product(
-                gs.matmul(inv_D_L, D_X), gs.matmul(inv_D_L, D_Y)
-            )
+    def test_inverse_differential_gram_belongs(self, n, tangent_vec, base_point):
+        result = self.space(n).inverse_differential_gram(
+            gs.array(tangent_vec), gs.array(base_point)
         )
-        ip_shape_result = ip_result.shape[0]
-        ip_shape_expected = n_samples
-
-        self.assertAllClose(ip_expected, ip_result)
-        self.assertAllClose(ip_shape_expected, ip_shape_result)
-
-    def test_exp(self):
-        """Test exp map"""
-        L = gs.array([[1.0, 0.0], [2.0, 2.0]])
-        X = gs.array([[-1.0, 0.0], [2.0, 3.0]])
-        exp_expected = gs.array([[1 / EULER, 0.0], [4.0, 2 * gs.exp(1.5)]])
-        exp_result = self.metric_cholesky.exp(X, L)
-        self.assertAllClose(exp_expected, exp_result)
-
-    def test_exp_vectorization(self):
-        """Test exp map vectorization"""
-        L = self.space.random_point(5)
-        X = self.space.ambient_space.random_point(5)
-        D_L = Matrices.to_diagonal(L)
-        D_X = Matrices.to_diagonal(X)
-        inv_D_L = gs.linalg.inv(D_L)
-        exp_expected = (
-            Matrices.to_strictly_lower_triangular(L)
-            + Matrices.to_strictly_lower_triangular(X)
-            + gs.matmul(D_L, SPDMatrices(2).expm(gs.matmul(D_X, inv_D_L)))
+        self.assertAllClose(gs.all(self.space(n).ambient_space.belongs(result)), True)
+
+
+class TestCholeskyMetric(TestCase, metaclass=Parametrizer):
+    cls = CholeskyMetric
+    space = PositiveLowerTriangularMatrices
+
+    class TestDataCholeskyMetric(TestData):
+        def diag_inner_product_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    tangent_vec_a=[[1.0, 0.0], [-2.0, -1.0]],
+                    tangent_vec_b=[[2.0, 0.0], [-3.0, -1.0]],
+                    base_point=[[SQRT_2, 0.0], [-3.0, 1.0]],
+                    expected=2.0,
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def strictly_lower_inner_product_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    tangent_vec_a=[[1.0, 0.0], [-2.0, -1.0]],
+                    tangent_vec_b=[[2.0, 0.0], [-3.0, -1.0]],
+                    expected=6.0,
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inner_product_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    tangent_vec_a=[[1.0, 0.0], [-2.0, -1.0]],
+                    tangent_vec_b=[[2.0, 0.0], [-3.0, -1.0]],
+                    base_point=[[SQRT_2, 0.0], [-3.0, 1.0]],
+                    expected=8.0,
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def exp_data(self):
```

Comment:
How about `exp_vectorization`?

Reply:
Ditto

Reply:
@ninamiolane it was not acutal batch vectorization data, but rewritten in simpler terms. I removed that, you can exp_belongs data